src/vis/Vis.tsx

```diff
@@ -0,0 +1,116 @@
+import d3 from 'd3';
+import * as React from 'react';
+import {useMemo, useState} from 'react';
+import {barMergeDefaultConfig, isBar} from './bar/utils';
+import {ENumericalColorScaleType, isScatter, scatterMergeDefaultConfig} from './scatter/utils';
+import {CategoricalColumn, NumericalColumn, ESupportedPlotlyVis, IVisConfig, Scales} from './interfaces';
+import {ScatterVis} from './scatter/ScatterVis';
+import {ViolinVis} from './violin/ViolinVis';
+import {isViolin, violinMergeDefaultConfig} from './violin/utils';
+import {isStrip, stripMergeDefaultConfig} from './strip/utils';
+import {StripVis} from './strip/StripVis';
+import {isPCP, pcpMergeDefaultConfig} from './pcp/utils';
+import {PCPVis} from './pcp/PCPVis';
+import {BarVis} from './bar/BarVis';
+
+export interface VisProps {
+    columns: (NumericalColumn | CategoricalColumn)[];
+    selected?: {[key: number]: boolean};
+    colors?: string[];
+    shapes?: string[];
+    selectionCallback?: (s: number[]) => void;
+    filterCallback?: (s: string) => void;
+}
+
+export function Vis({
+    columns,
+    selected = {},
+    colors = ['#337ab7', '#ec6836', '#75c4c2', '#e9d36c', '#24b466', '#e891ae', '#db933c', '#b08aa6', '#8a6044', '#7b7b7b'],
```

Comment:
Please retrieve the colors using the `getCssValue()` function introduced with PR https://github.com/phovea/phovea_ui/pull/161.

Reply:
done